src/repository/http.rs

```diff
@@ -228,10 +228,10 @@ where
             if status == StatusCode::NOT_FOUND {
                 Err(Error::NotFound)
             } else {
-                Err(Error::Opaque(format!(
-                    "Error getting {:?}: {:?}",
-                    self.uri, resp
-                )))
+                Err(Error::BadHttpStatus {
+                    code: status,
+                    uri: self.uri.to_string(),
```

Comment:
this field's documentation indicates this is the URI that was being fetched, but `self.uri` is just the base URI.  The local var `uri` had the full URI being requested.